pkg/server/ptypes/project_template_test.go

```diff
@@ -0,0 +1,153 @@
+package ptypes
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"testing"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+func TestValidateCreateProjectTemplateRequest(t *testing.T) {
+	tests := []struct {
+		name    string
+		req     *pb.CreateProjectTemplateRequest
+		wantErr bool
+	}{
+		{
+			name: "minimum valid request",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{
+					Name: "test",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "enforces name",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Inherits base validator rules (example: name length)",
+			req: &pb.CreateProjectTemplateRequest{
+				ProjectTemplate: &pb.ProjectTemplate{
+					Name: string(make([]byte, PROJECT_TEMPLATE_NAME_LENGTH+1)),
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ValidateCreateProjectTemplateRequest(tt.req)
+			if err != nil {
+				if !tt.wantErr {
+					t.Errorf("ValidateCreateProjectTemplateRequest() error = %v, wantErr %v", err, tt.wantErr)
+				}
+			}
```

Comment:
```suggestion
			if err == nil && tt.wantErr {
				t.Errorf("expected error in ValidateCreateProjectTemplateRequest() but got none")
			}

			if err != nil && !tt.wantErr {
				t.Errorf("ValidateCreateProjectTemplateRequest() error = %v, wantErr %v", err, tt.wantErr)
			}

```

Reply:
We should check `tt.wantErr` even if `err` is `nil`, or we miss situations where the error is nil but we expected one. 

For example, I cloned locally and added this test expecting it to fail:

```
{
        name: "enforces name to not be empty",
        req: &pb.CreateProjectTemplateRequest{
                ProjectTemplate: &pb.ProjectTemplate{
                        Name: " "},
        },
        wantErr: true,
},
```

However not only does that test case pass, I found that if I changed `wantErr` to `false` it still passes... I'll add that as an optional change. I'm not sure if we should consider that an error. I feel like we should, but I'm not sure if we guard against "empty" strings like that elsewhere